ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/datepicker/v1/datepicker/clientlibs/site/js/datepickerwidget.js

```diff
@@ -1081,7 +1084,9 @@ class DatePickerWidget {
         }
         if (this.#curInstance != null) {
             this.#curInstance.selectedDate = value;
-            this.#model.value = this.toString(); // updating model value to the latest
+            if(value) {
+                this.#model.value = this.toString(); // updating model value to the latest
+            }
```

Comment:
Setting the model inside setValue of widget is wrong approach. I've raised a new PR

Reply:
https://github.com/adobe/aem-core-forms-components/pull/816